src/traits.jl

```diff
@@ -46,3 +46,13 @@ struct ManyToMany <: Cardinality end
 Returns the [`Cardinality`](@ref) of the `transform`.
 """
 function cardinality end
+
+Base.:(∘)(::OneToOne, ::OneToOne) = OneToOne()
```

Comment:
I'm not sure if `∘` is the right syntax here... it's used to compose functions whereas here we're sort of "reducing" over the cardinalities.
I guess it makes sense in the context of composing the equivalent transforms?

If we do go for it then `∘` becomes part of the `Cardinality` API and should be documented.

Reply:
The alternative is to have an internal `_compose` function which would do this instead? I don't think users need to use this at all.

Reply:
yeah making it internal would indicate it shouldn't be used publically